src/mechanics/timeIntegration/NewmarkDirect.cpp

```diff
@@ -1,499 +1,182 @@
-#include "mechanics/dofSubMatrixStorage/BlockFullMatrix.h"
 #include "base/CallbackInterface.h"
 #include "base/Timer.h"
 #include "mechanics/structures/StructureOutputBlockMatrix.h"
-#include "mechanics/nodes/NodeBase.h"
-#include "mechanics/nodes/NodeEnum.h"
-#include "mechanics/groups/Group.h"
 #include "mechanics/structures/StructureBase.h"
-#include "mechanics/timeIntegration/ResultGroupNodeForce.h"
-#include "mechanics/timeIntegration/TimeIntegrationEnum.h"
+#include "mechanics/timeIntegration/postProcessing/ResultGroupNodeForce.h"
 #include "mechanics/constitutive/ConstitutiveEnum.h"
 #include "mechanics/constitutive/inputoutput/ConstitutiveCalculateStaticData.h"
 #include "mechanics/constitutive/inputoutput/ConstitutiveIOMap.h"
 #include "mechanics/constitutive/inputoutput/ConstitutiveScalar.h"
-#include "mechanics/elements/ElementBase.h"
 #include "mechanics/structures/StructureBaseEnum.h"
+#include "mechanics/timeIntegration/postProcessing/PostProcessor.h"
 #include "mechanics/timeIntegration/NewmarkDirect.h"
 #include "math/SparseMatrix.h"
 #include "math/SparseMatrixCSRVector2General.h"
 #include "mechanics/structures/Assembler.h"
 
-//! @brief constructor
-//! @param mDimension number of nodes
-NuTo::NewmarkDirect::NewmarkDirect(StructureBase* rStructure)
-    : NewmarkBase(rStructure)
+using namespace NuTo;
 
+NewmarkDirect::NewmarkDirect(StructureBase* rStructure)
+    : TimeIntegrationBase(rStructure)
+    , mHessian2(rStructure->GetDofStatus())
+    , mDampingMatrix(rStructure->GetDofStatus())
 {
 }
 
 
-//! @brief ... Info routine that prints general information about the object (detail according to verbose level)
-void NuTo::NewmarkDirect::Info() const
+void NewmarkDirect::Solve(double rTimeDelta)
 {
-    NewmarkBase::Info();
-}
-
-void NuTo::NewmarkDirect::Solve(double rTimeDelta)
-{
-    NuTo::Timer timerFull(__FUNCTION__, mStructure->GetShowTime(), mStructure->GetLogger());
+    mTimeControl.SetTimeFinal(rTimeDelta);
```

Comment:
Well rTimeDelta is somehow misleading. It should be renamed to rTimeFinal or something like this. We just didn't change it. So you provide the time where you like to stop when you call solve.